ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/LgrDB.hs

```diff
@@ -472,8 +473,11 @@ garbageCollectPrevApplied LgrDB{..} slotNo = modifyTVar varPrevApplied $
 
 -- | Wrap exceptions that may indicate disk failure in a 'ChainDbFailure'
 -- exception using the 'LgrDbFailure' constructor.
-wrapFailure :: forall m x. MonadCatch m => m x -> m x
-wrapFailure k = catch k rethrow
+wrapFailure :: forall m x blk. (MonadCatch m, Typeable blk, StandardHash blk) =>
+     Proxy blk
+  -> m x
+  -> m x
+wrapFailure _ k = catch k rethrow
   where
     rethrow :: FsError -> m x
-    rethrow err = throwIO $ LgrDbFailure err
+    rethrow err = throwIO $ (LgrDbFailure err :: ChainDbFailure blk)
```

Comment:
We prefer `-XTypeApplications` in such cases:
```suggestion
    rethrow err = throwIO $ LgrDbFailure @blk err
```
(The same comment applies to a bunch of other places)